be/src/exec/pipeline/scan/morsel.cpp

```diff
@@ -224,8 +261,11 @@ Segment* PhysicalSplitMorselQueue::_cur_segment() {
 }
 
 bool PhysicalSplitMorselQueue::_is_last_split_of_current_morsel() {
-    return _has_init_any_segment && _cur_segment() != nullptr && _cur_segment()->num_rows() != 0 &&
-           !_segment_range_iter.has_more();
+    if (_tablet_idx >= _tablet_rowsets.size()) {
+        return true;
+    }
+    return _num_segment_rest_rows == 0 && _rowset_idx + 1 == _tablet_rowsets[_tablet_idx].size() &&
```

Comment:
ensure TestQueryCacheSsb1gLineorderIntraTabletParallelism.test_query_cache_ssb1g_lineorder_intra_tablet_parallelism passed

Reply:
Yes, it passed.